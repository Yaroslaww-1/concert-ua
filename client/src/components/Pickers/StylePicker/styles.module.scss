@import 'src/styles/variables';

.root {
  max-height: 900px;
  width: 400px;
  max-width: 400px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  color: $colorBlack;
  margin: 20px 40px 20px 40px;

  @media (max-width: $desktopBreakpoint) {
    overflow-x: scroll;
    height: 600px;

    &::-webkit-scrollbar {
      height: 4px;
    }
  
    &::-webkit-scrollbar-thumb {
      background: $colorLightGray;
    }
  }

  @media (max-height: 900px) {
    overflow-x: scroll;
    height: 400px;
  }
}

.confirmingButtons {
  margin-left: 10%;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  padding: 0px 20px 20px 0px;
}