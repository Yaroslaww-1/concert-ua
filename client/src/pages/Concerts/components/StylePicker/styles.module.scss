@import 'src/styles/variables';

$iconSize: 20px;

.root {
  max-height: 600px;
  width: 400px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  color: $colorBlack;
  margin: 20px 40px 20px 40px;

  * {
    color: $colorBlack;
  }
}

.checkboxRoot {
  &:hover {
    background-color: transparent;
  }

  .icon {
    width: $iconSize;
    height: $iconSize;
    box-shadow: inset 0 0 0 0.5px rgba(16,22,26,.2), inset 0 -0.5px 0 rgba(16,22,26,.1);
    background-color: $colorWhite;
    input {
      &:hover {
        background-color: $colorRed,
      }
    };
  }

  .checkedIcon {
    background-color: $colorRed;
    box-shadow: none;

    &:before {
      color: transparent;
      display: block;
      width: $iconSize;
      height: $iconSize;
      background-image: url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'%3E%3Cpath fill-rule='evenodd' clip-rule='evenodd' d='M12 5c-.28 0-.53.11-.71.29L7 9.59l-2.29-2.3a1.003 1.003 0 00-1.42 1.42l3 3c.18.18.43.29.71.29s.53-.11.71-.29l5-5A1.003 1.003 0 0012 5z' fill='%23fff'/%3E%3C/svg%3E");
      content: '""';
    };
  }
}

.styleNameText {
  margin-left: 10px;
}

.confirmingButtons {
  margin-left: 10%;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  padding: 0px 20px 20px 0px;
}