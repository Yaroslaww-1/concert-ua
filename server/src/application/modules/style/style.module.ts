import { Module } from '@nestjs/common';
import { TypeOrmModule } from '@nestjs/typeorm';
import { StyleController } from './controllers/style.controller';
import { StyleEntity } from './entities/style.entity';
import { StyleRepository } from './repositories/style.repository';
import { StyleService } from './services/style.service';

const repositories = [StyleRepository];
const services = [StyleService];

@Module({
  imports: [TypeOrmModule.forFeature([StyleEntity])],
  controllers: [StyleController],
  providers: [...repositories, ...services],
})
export class StyleModule {}
