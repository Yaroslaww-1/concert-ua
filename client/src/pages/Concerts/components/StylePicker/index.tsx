import React from 'react';

import cssStyles from './styles.module.scss';

import { StyleModel } from 'src/api/models/style.model';
import BorderlessTransparentButton from 'src/components/Buttons/BorderlessTransparentButton';
import ColoredButton from 'src/components/Buttons/ColoredButton';
import CheckboxList from 'src/components/CheckboxList';

interface IProps {
  styles: StyleModel[];
  preSelected: StyleModel[];
  onClose: () => void;
  onSelect: (styles: StyleModel[]) => void;
}

const StylePicker: React.FC<IProps> = ({ styles, preSelected, onClose, onSelect }) => {
  const [selectedStyles, setSelectedStyles] = React.useState<StyleModel[]>(preSelected);

  const onToggle = (style: StyleModel, action: 'select' | 'unselect') => {
    if (action === 'select') {
      setSelectedStyles([...selectedStyles, style]);
    } else {
      setSelectedStyles(selectedStyles.filter((_style) => _style.id !== style.id));
    }
  };

  const onSubmitSelect = () => {
    onSelect(selectedStyles);
    onClose();
  };

  return (
    <>
      <CheckboxList<StyleModel>
        items={styles}
        preSelected={preSelected}
        onToggle={onToggle}
        classes={{ root: cssStyles.root }}
      />
      <div className={cssStyles.confirmingButtons}>
        <BorderlessTransparentButton onClick={onClose} text="Cancel" color="red" />
        <ColoredButton onClick={onSubmitSelect} text="Submit" />
      </div>
    </>
  );
};

export default StylePicker;
