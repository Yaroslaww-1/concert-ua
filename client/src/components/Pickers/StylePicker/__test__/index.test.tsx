import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import StylePicker from '../index';
import { defaultStyle, StyleModel } from 'src/api/models/style.model';

describe('StylePicker', () => {
  const setup = ({
    styles = [],
    preSelected = [],
    onClose = () => {},
    onSelect = () => null,
  }: {
    styles?: StyleModel[];
    preSelected?: StyleModel[];
    onClose?: () => void;
    onSelect?: () => Error | null;
  }) => {
    const { container } = render(
      <StylePicker styles={styles} preSelected={preSelected} onClose={onClose} onSelect={onSelect} />,
    );
    const closeButtonWrapper = container.getElementsByTagName('div').item(1);
    const closeButton = closeButtonWrapper?.getElementsByTagName('div').item(1)!;

    const stylesCheckboxes = container
      .getElementsByClassName('MuiFormGroup-root')
      .item(0)!
      .getElementsByTagName('label');

    const selectedStylesCheckboxes = container
      .getElementsByClassName('MuiFormGroup-root')
      .item(0)!
      .getElementsByClassName('Mui-checked');

    return {
      container,
      closeButton,
      stylesCheckboxes,
      selectedStylesCheckboxes,
    };
  };

  it('should match snapshot', () => {
    const { container } = setup({});
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call onClose on close', () => {
    const onClose = jest.fn(() => {});
    const { closeButton } = setup({ onClose });
    fireEvent.click(closeButton, onClose);
    expect(onClose).toHaveBeenCalled();
    expect(onClose).toHaveBeenCalledTimes(1);
  });

  it('should render styles checkboxes', () => {
    const { stylesCheckboxes } = setup({ styles: [defaultStyle] });
    expect(stylesCheckboxes.length).toBe(1);
  });

  it('should render styles checkboxes', () => {
    const { stylesCheckboxes } = setup({ styles: [defaultStyle] });
    expect(stylesCheckboxes.length).toBe(1);
  });

  it('should preselect styles checkboxes', () => {
    const { selectedStylesCheckboxes } = setup({ styles: [defaultStyle], preSelected: [defaultStyle] });
    expect(selectedStylesCheckboxes.length).toBe(1);
  });
});
