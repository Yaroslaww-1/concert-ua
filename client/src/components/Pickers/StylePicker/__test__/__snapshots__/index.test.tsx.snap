// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StylePicker should match snapshot 1`] = `
<div
  class="MuiFormGroup-root root root"
/>
`;
